packages/connections/gym/connection.py

```diff
@@ -32,7 +33,7 @@
 from aea.helpers.base import locate
 from aea.mail.base import Envelope
 
-if TYPE_CHECKING:
+if TYPE_CHECKING or "pytest" in sys.modules:
```

Comment:
this is the only way I could figure out how to check whether we are testing, not neat, but the whole import situation is not neat and needs to be resolved at some point

Reply:
I guess this is needed if you actually want to import the package module from outside the framework. For example, if you want to test a single class/method of a resource module.
However, this is not necessarily needed for every skill. Sometimes just integration test can be enough (that is, running the agent with the skill under testing though CLI command using different configurations.)

Reply:
true, we could rely on integration tests only for packages